### DISCUSSION TITLE
Can not install Kohya despite having Github installed.

### DISCUSSION THREAD

**Discussion by Deejay85** [bmaltais/kohya_ss#2373] [Q&A]
I updated Python to 10.9, and was finally able to get some of the installation to work, but apparently github is still acting up. I ran the install again, and got the following message:

```
07:52:34-338046 INFO     Kohya_ss GUI version: v24.0.6
07:52:34-343904 INFO     Python version is 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit
                         (AMD64)]
fatal: not a git repository (or any of the parent directories): .git
07:52:34-581194 ERROR    Error during Git operation: Command '['git', 'submodule', 'update', '--init', '--recursive',
                         '--quiet']' returned non-zero exit status 128.
07:52:34-584124 INFO     Installing requirements from requirements_pytorch_windows.txt...
Looking in indexes: https://download.pytorch.org/whl/cu118

Requirement already satisfied: torch==2.1.2+cu118 in m:\kohya_ss\venv\lib\site-packages (from -r requirements_pytorch_windows.txt (line 1)) (2.1.2+cu118)
Requirement already satisfied: torchvision==0.16.2+cu118 in m:\kohya_ss\venv\lib\site-packages (from -r requirements_pytorch_windows.txt (line 2)) (0.16.2+cu118)
Requirement already satisfied: xformers==0.0.23.post1+cu118 in m:\kohya_ss\venv\lib\site-packages (from -r requirements_pytorch_windows.txt (line 3)) (0.0.23.post1+cu118)
.
.
.
[notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
07:52:35-097766 INFO     Requirements from requirements_pytorch_windows.txt installed.
07:52:35-099719 INFO     Installing requirements from requirements_windows.txt...
Obtaining file:///M:/kohya_ss/sd-scripts (from -r requirements.txt (line 35))
ERROR: file:///M:/kohya_ss/sd-scripts (from -r requirements.txt (line 35)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

[notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
07:52:35-506920 INFO     Requirements from requirements_windows.txt installed.
'accelerate' is not recognized as an internal or external command,
operable program or batch file.[notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
07:52:35-097766 INFO     Requirements from requirements_pytorch_windows.txt installed.
07:52:35-099719 INFO     Installing requirements from requirements_windows.txt...
Obtaining file:///M:/kohya_ss/sd-scripts (from -r requirements.txt (line 35))
ERROR: file:///M:/kohya_ss/sd-scripts (from -r requirements.txt (line 35)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

[notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
07:52:35-506920 INFO     Requirements from requirements_windows.txt installed.
'accelerate' is not recognized as an internal or external command,
operable program or batch file.
```

As you can see the Git part is acting up for some odd reason. I already went to https://desktop.github.com, installed the latest version, and that didn't seem to fix it...am I doing something wrong? Also I checked the M:\kohya_ss\sd-scripts folder, and it appears to be empty...which makes sense since that was how it was before the installation even took place.

**Comment by Serallan** [comment]
First upgrade your pip like the message says. Pip will be used to install the packages in requirements.
```
python.exe -m pip install --upgrade pip
```

About git, don't forget to cd into the dir relevant to it.

I don't know what's up with your installation, but you can get sd-scripts manually.

```
# Get into kohya folder, where sd-scripts will be copied to
cd /path/to/kohya

# Get scripts dir
# Add "-b vSomeVersion" if you want to get the branch of a specific version, eg v0.8.7, otherwise it'll get the latest
git clone -b v0.8.7 https://github.com/kohya-ss/sd-scripts

# You can also do a checkout to get to a release from its hash
# First change the working directory to sd-scripts, which you'll be modifying with the checkout
cd /path/to/kohya/sd-scripts

# Eg hash bfb352b (same as v0.8.7)
git checkout bfb352b
```

Now install the sd-scripts requirements, so execute the txt from that folder.

**Reply by Deejay85** [reply]
I tried to upgrade the PIP before, but it only upgraded the one installed on my OS, and not the one for the instance of Kohya that I'm running. I did try running Python.exe in the venv\Scripts folder, but all I got was an error message:

```
  File "<stdin>", line 1
    python.exe -m pip install --upgrade pip
                  ^^^
SyntaxError: invalid syntax
```

Did copy the latest version to the sd-scripts folder, and that didn't seem to make any difference...also how am I to install the stuff in the requirements.txt when there doesn't seem to be anything to extract the list form it? Am I supposed to replace the requirements.txt file in the root directory, and install it from there?

**Reply by Serallan** [reply]
You need to get python and pip working before doing anything else. Forget kohya reqs by now. Which version of python you got at venv?

Is that pip error the complete error message you're getting?

**Comment by Deejay85** [comment]
The version is 3.10.9. As for the pip error, yes, that is the complete message. Above it is the command I entered, and beneath it is a new line waiting for additional input. Why is the pip command the one being highlighted? Not a clue. When I enter the command into a traditonal CMD window it says that requirement has already been satisfied, which while nice, does nothing for the one in venv.

EDIT: Well this is odd...I went to exit out of setup.bat by hitting 7 to exit, and instead of the window closing, it was still open, but still connected to the venv version of Python. So I entered the update command and it worked! Can't say I saw that coming. So the good news is that since PIP is now updated, I ran the install Kohyaa GUI, got some stuff downloaded, and now all that's left to fix is my original issue,

```
12:59:27-895401 ERROR    Error during Git operation: Command '['git', 'submodule', 'update', '--init', '--recursive',
                         '--quiet']' returned non-zero exit status 128.
```

 and trying to figuring out what to do with/how to install the files in my sd-scripts folder...small successes. 🎉

**Reply by DoodleSurf** [reply]
`fatal: not a git repository (or any of the parent directories): .git`
I had the same issue, realized that the Releases packages don't incluide `.git` folder which links to the other github repo (git submodule) with the sd-scripts folder.

Cloning the repo with `git clone` and then installing the gui should alleviate an issue.

**Reply by AlexiosDyral** [reply]
i had the same issue,and i use `git clone -b v0.8.7 https://github.com/kohya-ss/sd-scripts` .to solve problem of “the sd-scripts”.
how can i use `git clone` to get a git repository?

**Reply by AlexiosDyral** [reply]
fine, i got the solution
